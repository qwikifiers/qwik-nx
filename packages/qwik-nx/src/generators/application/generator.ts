import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { NormalizedSchema, normalizeOptions } from './utils/normalize-options';
import { viteConfigurationGenerator } from '@nrwl/vite';
import { QwikAppGeneratorSchema } from './schema';
import { Linter } from '@nrwl/linter';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { configureEslint } from '../../utils/configure-eslint';
import { addCommonQwikDependencies } from '../../utils/add-common-qwik-dependencies';

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    template: '',
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.appProjectRoot,
    templateOptions
  );
}

async function configureVite(
  tree: Tree,
  options: NormalizedSchema
): Promise<GeneratorCallback> {
  const tsConfigPath = joinPathFragments(
    options.appProjectRoot,
    'tsconfig.json'
  );
  const tsConfig = tree.read(tsConfigPath);
  const includeVitest = options.unitTestRunner === 'vitest';

  const callback = await viteConfigurationGenerator(tree, {
    project: options.projectName,
    newProject: false,
    includeVitest,
    includeLib: false,
    uiFramework: 'none',
    inSourceTests: false,
  });
  tree.delete(joinPathFragments(options.appProjectRoot, 'index.html'));

  // overwrite tsconfig back after "viteConfigurationGenerator"
  tree.write(tsConfigPath, tsConfig);

  if (includeVitest) {
    const projectConfig = readProjectConfiguration(tree, options.projectName);
    const testTarget = projectConfig.targets['test'];
    testTarget.outputs = ['{workspaceRoot}/coverage/{projectRoot}'];
    updateProjectConfiguration(tree, options.projectName, projectConfig);
  }

  return callback;
}

export default async function (tree: Tree, options: QwikAppGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);
  const tasks: GeneratorCallback[] = [];

  addProjectConfiguration(tree, normalizedOptions.projectName, {
    root: normalizedOptions.appProjectRoot,
    name: normalizedOptions.projectName,
    projectType: 'application',
    sourceRoot: `${normalizedOptions.appProjectRoot}/src`,
    // TODO: revisit later, now "viteConfigurationGenerator" will fail if there's no build target specified
    targets: { build: { options: {} } },
    tags: normalizedOptions.parsedTags,
  });

  addFiles(tree, normalizedOptions);

  tasks.push(await configureVite(tree, normalizedOptions));

  if (normalizedOptions.linter === Linter.EsLint) {
    tasks.push(configureEslint(tree, normalizedOptions.projectName, true));
  }

  if (normalizedOptions.style !== 'none') {
    tasks.push(
      addStyledModuleDependencies(tree, normalizedOptions.styleExtension)
    );
  }

  tasks.push(addCommonQwikDependencies(tree));

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}
