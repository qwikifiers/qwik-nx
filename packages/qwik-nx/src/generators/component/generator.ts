import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  getProjects,
  joinPathFragments,
  logger,
  names,
  Tree,
} from '@nrwl/devkit';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { ComponentGeneratorSchema } from './schema';

interface NormalizedSchema extends ComponentGeneratorSchema {
  directory: string;
  hasStyles: boolean;
  projectRoot: string;
}

function getDirectory(host: Tree, options: ComponentGeneratorSchema) {
  const workspace = getProjects(host);
  let baseDir: string;
  if (options.directory) {
    baseDir = options.directory;
  } else {
    baseDir =
      workspace.get(options.project)!.projectType === 'application'
        ? 'components'
        : 'lib';
  }
  return options.flat
    ? baseDir
    : joinPathFragments(baseDir, names(options.name).fileName);
}

function normalizeOptions(
  host: Tree,
  options: ComponentGeneratorSchema
): NormalizedSchema {
  const project = getProjects(host).get(options.project);

  if (!project) {
    logger.error(
      `Cannot find the ${options.project} project. Please double check the project name.`
    );
    throw new Error();
  }

  const projectRoot =
    project.sourceRoot ?? joinPathFragments(project.root, 'src');

  const directory = getDirectory(host, options);

  return {
    ...options,
    directory,
    hasStyles: options.style !== 'none',
    projectRoot,
  };
}

function createComponentFiles(tree: Tree, options: NormalizedSchema) {
  const libNames = names(options.name);
  const hasStyles = options.style && options.style !== 'none';
  const templateOptions = {
    ...options,
    ...libNames,
    hasStyles,
  };

  const componentDir = joinPathFragments(
    options.projectRoot,
    options.directory
  );

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files/common'),
    componentDir,
    templateOptions
  );
  if (hasStyles) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, 'files/styles'),
      componentDir,
      templateOptions
    );
  }
  if (!options.skipTests) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, 'files/tests'),
      componentDir,
      templateOptions
    );
  }
}

export async function componentGenerator(
  tree: Tree,
  options: ComponentGeneratorSchema
): Promise<GeneratorCallback> {
  const normalizedOptions = normalizeOptions(tree, options);
  createComponentFiles(tree, normalizedOptions);
  await formatFiles(tree);

  return addStyledModuleDependencies(tree, normalizedOptions.style);
}

export default componentGenerator;
