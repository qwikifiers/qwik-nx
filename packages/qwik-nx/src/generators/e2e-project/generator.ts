import {
  ensurePackage,
  GeneratorCallback,
  readProjectConfiguration,
  Tree,
} from '@nrwl/devkit';
import { getInstalledNxVersion } from '../../utils/get-installed-nx-version';
import { nxKitVersion } from '../../utils/versions';
import { E2eProjectGeneratorSchema, NormalizedSchema } from './schema';
import { normalizeOptions } from './utils/normalize-options';

export async function addE2eProject(
  tree: Tree,
  options: E2eProjectGeneratorSchema
): Promise<GeneratorCallback> {
  const projectConfiguration = readProjectConfiguration(tree, options.project);

  if (projectConfiguration.projectType !== 'application') {
    throw new Error('Cannot setup e2e project for the given frontend project.');
  }

  const normalizedOptions = normalizeOptions(tree, options);
  if (options.e2eTestRunner === 'cypress') {
    return addCypress(tree, normalizedOptions);
  }

  if (options.e2eTestRunner === 'playwright') {
    return addPlaywright(tree, normalizedOptions);
  }

  return () => void 0;
}

async function addCypress(tree: Tree, options: NormalizedSchema) {
  await ensurePackage(tree, '@nrwl/cypress', getInstalledNxVersion(tree));
  const { cypressProjectGenerator } = await import('@nrwl/cypress');

  return await cypressProjectGenerator(tree, {
    ...options,
    name: options.e2eProjectName,
    directory: options.directory,
    project: options.project,
    bundler: 'vite',
    skipFormat: options.skipFormat,
  });
}

async function addPlaywright(tree: Tree, options: NormalizedSchema) {
  await ensurePackage(tree, '@nxkit/playwright', nxKitVersion);
  const { projectGenerator } = await import('@nxkit/playwright');

  return await projectGenerator(tree, {
    ...options,
    name: options.e2eProjectName,
    directory: options.directory,
    frontendProject: options.project,
    skipFormat: options.skipFormat,
  });
}

export default addE2eProject;
