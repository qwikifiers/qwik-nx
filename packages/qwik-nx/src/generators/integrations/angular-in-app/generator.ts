import { Tree } from '@nx/devkit';
import { AngularInAppGeneratorSchema } from './schema';
import { ensureNxDependencies } from './../../../utils/ensure-nx-dependencies';

export async function angularInAppGenerator(
  tree: Tree,
  schema: AngularInAppGeneratorSchema
) {
  ensureNxDependencies();

  const { angularInAppGeneratorInternal } = await import(
    './generator-internal'
  );
  return angularInAppGeneratorInternal(tree, schema);
}

export default angularInAppGenerator;
