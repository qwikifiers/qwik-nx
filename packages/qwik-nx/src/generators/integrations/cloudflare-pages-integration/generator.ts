import {
  addDependenciesToPackageJson,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  offsetFromRoot,
  ProjectConfiguration,
  readProjectConfiguration,
  TargetConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import {
  getIntegrationConfigurationName,
  IntegrationName,
} from '../../../utils/integration-configuration-name';
import { nxCloudflareWrangler, wranglerVersion } from '../../../utils/versions';
import { CloudflarePagesIntegrationGeneratorSchema } from './schema';

interface NormalizedOptions {
  offsetFromRoot: string;
  projectConfig: ProjectConfiguration;
}

export async function cloudflarePagesIntegrationGenerator(
  tree: Tree,
  options: CloudflarePagesIntegrationGeneratorSchema
) {
  const config = readProjectConfiguration(tree, options.project);
  if (config.projectType !== 'application') {
    throw new Error(
      'Cannot setup cloudflare integration for the given project.'
    );
  }
  if (config.targets?.['build']?.executor !== 'qwik-nx:build') {
    throw new Error(
      'Project contains invalid configuration. ' +
        'If you encounter this error within a Qwik project, make sure you have run necessary Nx migrations for qwik-nx plugin.'
    );
  }

  const configurationName = getIntegrationConfigurationName(
    IntegrationName.Cloudflare,
    config
  );
  const deployTargetName = config.targets['deploy']
    ? 'deploy.cloudflare'
    : 'deploy';

  const normalizedOptions = normalizeOptions(config);
  (config.targets['build']?.configurations ?? {})[configurationName] = {};
  (config.targets['build.ssr'].configurations ??= {})[configurationName] =
    getBuildSSRTargetCloudflareConfiguration(normalizedOptions);
  config.targets[deployTargetName] = getDeployTarget(normalizedOptions);
  config.targets['preview-cloudflare'] =
    getCloudflarePreviewTarget(normalizedOptions);
  config.targets['build-cloudflare'] = getIntermediateDependsOnTarget(
    normalizedOptions,
    configurationName
  );

  updateProjectConfiguration(tree, options.project, config);

  addFiles(tree, normalizedOptions);

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return addCloudflarePagesDependencies(tree);
}

function getBuildSSRTargetCloudflareConfiguration(options: NormalizedOptions) {
  return {
    configFile: `${options.projectConfig.root}/adapters/cloudflare-pages/vite.config.ts`,
  };
}

function getDeployTarget(options: NormalizedOptions): TargetConfiguration {
  return {
    executor: '@k11r/nx-cloudflare-wrangler:deploy-page',
    options: {
      dist: `dist/${options.projectConfig.root}/client`,
    },
    dependsOn: ['build-cloudflare'],
  };
}

function getCloudflarePreviewTarget(
  options: NormalizedOptions
): TargetConfiguration {
  return {
    executor: '@k11r/nx-cloudflare-wrangler:serve-page',
    options: {
      dist: `dist/${options.projectConfig.root}/client`,
    },
    dependsOn: ['build-cloudflare'],
  };
}

/** Currently it's not possible to depend on a target with a specific configuration, that's why intermediate one is required */
function getIntermediateDependsOnTarget(
  options: NormalizedOptions,
  configurationName: string
): TargetConfiguration {
  return {
    executor: 'nx:run-commands',
    options: {
      command: `npx nx run ${options.projectConfig.name}:build:${configurationName}`,
    },
  };
}

function normalizeOptions(
  projectConfig: ProjectConfiguration
): NormalizedOptions {
  return {
    projectConfig,
    offsetFromRoot: offsetFromRoot(projectConfig.root),
  };
}

function addFiles(tree: Tree, options: NormalizedOptions): void {
  const templateOptions = {
    ...names(options.projectConfig.name!),
    projectRoot: options.projectConfig.root,
    offsetFromRoot: options.offsetFromRoot,
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.projectConfig.root,
    templateOptions
  );
}

function addCloudflarePagesDependencies(tree: Tree): GeneratorCallback {
  return addDependenciesToPackageJson(
    tree,
    {},
    {
      wrangler: wranglerVersion,
      '@k11r/nx-cloudflare-wrangler': nxCloudflareWrangler,
    }
  );
}

export default cloudflarePagesIntegrationGenerator;
