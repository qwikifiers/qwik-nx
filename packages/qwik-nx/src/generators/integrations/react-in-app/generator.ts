import {
  formatFiles,
  generateFiles,
  joinPathFragments,
  ProjectType,
  readProjectConfiguration,
  Tree,
} from '@nx/devkit';
import * as path from 'path';
import { ReactInAppGeneratorSchema } from './schema';
import { reactInit } from '../../../utils/react/init';

interface NormalizedSchema extends ReactInAppGeneratorSchema {
  sourceRoot: string;
  projectRoot: string;
  projectType: ProjectType;
}

function normalizeOptions(
  tree: Tree,
  options: ReactInAppGeneratorSchema
): NormalizedSchema {
  const projectConfig = readProjectConfiguration(tree, options.project);

  return {
    ...options,
    installMUIExample: options.installMUIExample !== false,
    sourceRoot: projectConfig.sourceRoot ?? projectConfig.root + '/src',
    projectRoot: projectConfig.root,
    projectType: projectConfig.projectType!,
  };
}

function addFiles(tree: Tree, normalizedOptions: NormalizedSchema): void {
  const filePath = normalizedOptions.installMUIExample ? 'mui' : 'demo';
  generateFiles(
    tree,
    path.join(__dirname, 'files', filePath),
    joinPathFragments(normalizedOptions.sourceRoot, 'routes/react'),
    {}
  );
}

export async function reactInAppGenerator(
  tree: Tree,
  schema: ReactInAppGeneratorSchema
) {
  const normalizedOptions = normalizeOptions(tree, schema);

  if (normalizedOptions.projectType !== 'application') {
    throw new Error(
      `Only applications are supported, "${normalizedOptions.project}" is a library.`
    );
  }

  const demoFilePath = joinPathFragments(
    normalizedOptions.sourceRoot,
    'integrations/react'
  );

  if (tree.exists(demoFilePath)) {
    throw new Error(
      `Looks like react integration has already been configured for ${normalizedOptions.project}. "${demoFilePath}" already exists.`
    );
  }

  const initCallback = reactInit(tree, {
    demoFilePath: joinPathFragments(
      normalizedOptions.sourceRoot,
      'integrations/react'
    ),
    installMUIExample: !!normalizedOptions.installMUIExample,
    projectRoot: normalizedOptions.projectRoot,
  });
  addFiles(tree, normalizedOptions);
  if (!normalizedOptions.skipFormat) {
    await formatFiles(tree);
  }

  return initCallback;
}

export default reactInAppGenerator;
