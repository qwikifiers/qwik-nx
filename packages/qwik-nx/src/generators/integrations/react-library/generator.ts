import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  output,
  readProjectConfiguration,
  Tree,
} from '@nrwl/devkit';
import * as path from 'path';
import { ReactLibraryGeneratorSchema } from './schema';
import { normalizeOptions as libraryNormalizeOptions } from './../../library/utils/normalize-options';
import libraryGenerator from '../../library/generator';
import {
  addReactPluginToViteConfig,
  reactInit,
} from '../../../utils/react/init';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';

interface NormalizedSchema extends ReactLibraryGeneratorSchema {
  installMUIExample: boolean;
  targetApps: string[];
  projectName: string;
  projectRoot: string;
  projectDirectory: string;
  parsedTags: string[];
}

function normalizeOptions(
  tree: Tree,
  options: ReactLibraryGeneratorSchema
): NormalizedSchema {
  const normalizedLibraryOptions = libraryNormalizeOptions(tree, {
    ...options,
    style: 'none',
  });

  let targetApps: string[] = [];
  if (Array.isArray(options.targetApps)) {
    targetApps = options.targetApps.filter(Boolean);
  } else if (
    typeof options.targetApps === 'string' &&
    options.targetApps.trim()
  ) {
    // can be of type "string" if value comes from the x-prompt
    targetApps = options.targetApps
      .split(',')
      .map((v) => v.trim())
      .filter(Boolean);
  }

  return {
    ...normalizedLibraryOptions,
    installMUIExample: options.installMUIExample !== false,
    targetApps,
  };
}

function addFiles(tree: Tree, options: NormalizedSchema) {
  generateFiles(
    tree,
    path.join(__dirname, 'files'),
    options.projectRoot,
    options
  );
}

function updateTargetApps(tree: Tree, targetApps: string[]) {
  if (!targetApps?.length) {
    output.warn({
      title: `You haven't provided any "targetApps" to be configured.`,
      bodyLines: [
        'Qwik React integration requires "qwikReact()" plugin to be included for all apps where qwikified react components are used.',
        'You can still import the plugin manually.',
      ],
    });
  } else {
    for (const targetApp of new Set(targetApps)) {
      const config = readProjectConfiguration(tree, targetApp);
      if (config.projectType !== 'application') {
        throw new Error(`"${targetApp}" project is not an application`);
      }
      addReactPluginToViteConfig(tree, config.root);
    }
  }
}

export async function reactLibraryGenerator(
  tree: Tree,
  schema: ReactLibraryGeneratorSchema
) {
  const normalizedOptions = normalizeOptions(tree, schema);
  const tasks: GeneratorCallback[] = [];

  const installTask = await libraryGenerator(tree, {
    ...normalizedOptions,
    skipFormat: true,
    generateComponent: false,
  });
  tasks.push(installTask);

  const initTask = reactInit(tree, {
    demoFilePath: joinPathFragments(
      normalizedOptions.projectRoot,
      'src/lib/integration'
    ),
    installMUIExample: !!normalizedOptions.installMUIExample,
    projectRoot: normalizedOptions.projectRoot,
  });
  tasks.push(initTask);

  addFiles(tree, normalizedOptions);

  updateTargetApps(tree, normalizedOptions.targetApps);

  if (!normalizedOptions.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}

export default reactLibraryGenerator;
