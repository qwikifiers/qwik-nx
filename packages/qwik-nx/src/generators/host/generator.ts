import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { HostGeneratorSchema } from './schema';
import { appGenerator } from './../application/generator';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import remoteGenerator from '../remote/generator';
import { normalizeOptions } from '../application/utils/normalize-options';
import { join } from 'path';
import { addMicroFrontendBetaWarning } from '../../utils/mf-beta-warning';
import { NormalizedSchema } from '../application/schema';

function addFiles(
  tree: Tree,
  schema: NormalizedSchema,
  remotes: { name: string; port: number }[]
) {
  // TODO: switch to empty preset once available
  tree.delete(
    joinPathFragments(schema.projectRoot, 'src/routes/flower/index.tsx')
  );
  tree.delete(
    joinPathFragments(schema.projectRoot, 'src/routes/flower/flower.css')
  );

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    schema.projectRoot,
    {
      name: schema.projectName,
      remotes: remotes.map((r) => r.name),
    }
  );

  tree.write(
    join(schema.projectRoot, 'src/config/remotes.json'),
    JSON.stringify(
      Object.fromEntries(
        remotes.map((r) => [r.name, `http://localhost:${r.port}`])
      )
    )
  );

  const projectConfig = readProjectConfiguration(tree, schema.projectName);
  projectConfig.targets!['serve'].executor =
    'qwik-nx:micro-frontends-dev-server';
  projectConfig.targets!['preview'].executor =
    'qwik-nx:micro-frontends-preview-server';
  updateProjectConfiguration(tree, schema.projectName, projectConfig);
}

export async function hostGenerator(tree: Tree, options: HostGeneratorSchema) {
  addMicroFrontendBetaWarning();
  const tasks: GeneratorCallback[] = [];

  const normalizedSchema = normalizeOptions(tree, {
    ...options,
    devServerPort: options.port,
    previewServerPort: options.port,
  });

  const initTask = await appGenerator(tree, {
    ...normalizedSchema,
    skipFormat: true,
  });
  tasks.push(initTask);

  const remotesWithPorts: {
    name: string;
    port: number;
  }[] = [];

  if (options.remotes) {
    let port = normalizedSchema.devServerPort + 1;
    for (const remote of options.remotes) {
      remotesWithPorts.push({
        name: remote,
        port,
      });

      await remoteGenerator(tree, {
        ...options,
        name: remote,
        port,
        skipFormat: true,
      });
      port++;
    }
  }

  addFiles(tree, normalizedSchema, remotesWithPorts);

  if (!options.skipFormat) {
    await formatFiles(tree);
  }
  return runTasksInSerial(...tasks);
}

export default hostGenerator;
