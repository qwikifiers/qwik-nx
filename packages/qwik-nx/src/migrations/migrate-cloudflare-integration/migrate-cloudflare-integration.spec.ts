import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree } from '@nx/devkit';

import update from './migrate-cloudflare-integration';
import { appGenerator } from '../../generators/application/generator';

describe('migrate-cloudflare-integration migration', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await appGenerator(tree, { name: 'myapp' });
    tree.write('apps/myapp/functions/[[path]].ts', 'export default {}');
    tree.write(
      'apps/myapp/adapters/cloudflare-pages/vite.config.ts',
      'export default {}'
    );
    tree.write(
      'apps/myapp/src/entry.cloudflare-pages.tsx',
      `import { createQwikCity } from '@builder.io/qwik-city/middleware/cloudflare-pages';
    import qwikCityPlan from '@qwik-city-plan';
    import render from './entry.ssr';
    
    const onRequest = createQwikCity({ render, qwikCityPlan });
    
    export { onRequest };`
    );
  });

  it('should run successfully', async () => {
    await update(tree);
    expect(tree.exists('apps/myapp/functions/[[path]].ts')).toBe(false);
    expect(
      tree.read('apps/myapp/src/entry.cloudflare-pages.tsx', 'utf-8')
    ).toMatchSnapshot('entry.cloudflare-pages.tsx');
  });
});
