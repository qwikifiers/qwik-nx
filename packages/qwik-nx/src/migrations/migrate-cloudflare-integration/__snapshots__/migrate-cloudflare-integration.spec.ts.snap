// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migrate-cloudflare-integration migration should run successfully: entry.cloudflare-pages.tsx 1`] = `
"import { createQwikCity } from '@builder.io/qwik-city/middleware/cloudflare-pages';
    import qwikCityPlan from '@qwik-city-plan';
    import render from './entry.ssr';
    
    const fetch = createQwikCity({ render, qwikCityPlan });
    
    export { fetch };"
`;
